Dynamic programming - algos/max_sub_square.cpp

```diff
@@ -0,0 +1,79 @@
+////////////////////////////Problem statement/////////////////////////////
+
+/////
+
+Given a binary matrix, find out the maximum size square sub-matrix with all 1s.
+
+Input:
+
+The first line of input contains an integer T denoting the number of test cases.
+The first line of each test case is n and m,n is the number of rows and m is the number of columns.
```

Comment:
there are m rows and n columns. ( Reference line 65 - 75, study line 73 carefully). Also edit the information then in line no. 10,11,21.